### PR TITLE
fix cargo errors

### PR DESCRIPTION
**Pull request by azjezz**

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -2,8 +2,10 @@
...
 
 [dependencies]
-error-chain = "0.12.4"
-reqwest = "0.11.11"
-serde_json = "1.0.82"
\ No newline at end of file
+reqwest = { version = "0.11", features = ["json"] }
+serde = { version = "1.0", features = ["derive"] }
+serde_json = { version = "1.0" }
+tokio = { version = "1", features = ["full"] }
```
Comment (azjezz) [inline Cargo.toml:11]: tokio is needed to make `main()` async :)

File: Cargo.toml [hunk 2]
```diff
@@ -2,8 +2,10 @@
 name = "send-with-sendgrid-and-rust"
 version = "0.1.0"
 authors = ["Matthew Setter"]
+edition = "2018"
```
Comment (azjezz) [inline Cargo.toml:5]: this was previously missing, which resulted in the "missing dependency" errors

File: src/main.rs [hunk 3]
```diff
@@ -3,24 +3,24 @@ use reqwest::Client;
...
 
-fn main() ->  Result<(), Err()> {
-    struct User {
-        name: String,
-        email: String
-    }
+struct User {
+    name: String,
+    email: String,
+}
```
Comment (azjezz) [inline src/main.rs:9]: just a preference, can be kept inside the main function

File: src/main.rs [hunk 4]
```diff
@@ -3,24 +3,24 @@ use reqwest::Client;
...
 
+#[tokio::main]
```
Comment (azjezz) [inline src/main.rs:11]: this attribute allows main() to be async.

File: src/main.rs [hunk 5]
```diff
@@ -51,16 +51,19 @@ fn main() ->  Result<(), Err()> {
...
+        .send()
+        .await
+        .unwrap();
 
     // Handle/Check the response
-    match response.status() {
+    match response.status().as_u16() {
         200 | 201 | 202 => println!("Email sent!"),
-        _ => println!("Unable to send your email"),
+        _ => eprintln!("Unable to send your email"),
```
Comment (azjezz) [inline src/main.rs:65]: eprintln prints to stderr instead of stdout

File: src/main.rs [hunk 6]
```diff
@@ -51,16 +51,19 @@ fn main() ->  Result<(), Err()> {
...
+        .json(&body)
+        .header(header::AUTHORIZATION, format!("Bearer {}", api_key))
+        .header(header::CONTENT_TYPE, header::HeaderValue::from_static("application/json"))
+        .send()
+        .await
+        .unwrap();
 
     // Handle/Check the response
-    match response.status() {
+    match response.status().as_u16() {
```
Comment (azjezz) [inline src/main.rs:63]: .status() returns a StatucCode, not an "integer", using "as_u116()" converts StatusCode into an integer ( u16 )

File: src/main.rs [hunk 7]
```diff
@@ -51,16 +51,19 @@ fn main() ->  Result<(), Err()> {
...
-        .default_headers(headers)
-        .json(&body)
         .post("https://api.sendgrid.com/v3/mail/send")
-        .send().await?;
+        .json(&body)
+        .header(header::AUTHORIZATION, format!("Bearer {}", api_key))
+        .header(header::CONTENT_TYPE, header::HeaderValue::from_static("application/json"))
+        .send()
+        .await
+        .unwrap();
```
Comment (azjezz) [inline src/main.rs:60]: P.S: this should probably handled better, the client could have failed to send the request.

File: Cargo.toml (hunk 2 above)
Comment (settermjd) [reply Cargo.toml:5]: I totally overlooked that in the documentation. Thank you.

File: src/main.rs (hunk 3 above)
Comment (settermjd) [reply src/main.rs:9]: Sweet! 👍🏽

File: src/main.rs (hunk 5 above)
Comment (settermjd) [reply src/main.rs:65]: Right! I remember reading about that, now that you mention it.

**Review by settermjd — APPROVED** [review]
👍🏽

**Pull request merged**